src/blueapi/core/context.py

```diff
@@ -106,6 +106,8 @@ def find_device(self, addr: str | list[str]) -> Device | None:
             return find_component(self.devices, addr)
 
     def with_config(self, config: EnvironmentConfig) -> None:
+        if config.metadata:
```

Comment:
Should: Be explicit:

```suggestion
        if config.metadata is not None:
```

Reply:
Done.